compareBuckets/compareBuckets.js

```diff
@@ -91,15 +87,14 @@ function compareBuckets(params, log, cb) {
                     if (srcDone || srcContents.length > 0) {
                         return process.nextTick(_done);
                     }
-
                     return listBucketMasterKeys(
                         bucketdSrcParams,
                         (err, isTruncated, marker, contents) => {
                             if (err) {
+                                log.error('Error fetching src bucket', { error: err });
                                 return _done(err);
                             }
-
-                            srcContents = contents;
+                            srcContents = contents || [];
```

Comment:
why this chnage?
contents cannot really be empty... esp. looking at `listBucketMasterKeys()`, we expect it to be returned, and access elements within anyway: so it would fail (crash) if no present.